File: thread_locks/c++/hello_thread.cpp

#include <iostream>
#include <thread>

using std::cout;
using std::endl;

void run() {
    cout << "Hello from new thread" << endl;
}

int main() {
    std::thread thread(run);
    cout << "Hello from main" << endl;
    thread.join();
    return 0;
}

File: thread_locks/clojure/hello_thread.clj
;; Simple 'hello' thread exemple.
;; Run me with 'clojure -cp . hello_thread.clj'

(defn hello
  []
  (println "Hello from new thread")
  )

(defn main
  "Main function"
  []
  (println "Hello from main")
  (.start (Thread. (hello)))
  )

(main)
